recipe-client-addon/lib/NormandyApi.jsm

```diff
@@ -49,8 +49,29 @@ this.NormandyApi = {
     return this.apiCall("post", endpoint, data);
   },
 
+  apiIndex() {
+    if (!indexPromise) {
+      indexPromise = this.get(prefs.getCharPref("api_url")).then(res => res.json);
+    }
+    return indexPromise;
+  },
+
+  makeApiUrl(endpoint) {
+    let base = prefs.getCharPref("api_url");
+    // Remove trailing slash if present
+    if (base.slice(-1) === "/") {
+      base = base.slice(0, -1);
+    }
+    // Remove leading slash, if present
+    if (endpoint[0] === "/") {
+      endpoint = endpoint.slice(1);
+    }
+    return `${base}/${endpoint}`;
+  },
+
   fetchRecipes: Task.async(function* (filters = {}) {
-    const recipeResponse = yield this.get("recipe/signed/", filters);
+    const signedRecipeUrls = (yield this.apiIndex())["recipes-signed"];
+    const recipeResponse = yield this.get(signedRecipeUrls, filters, {enabled: true});
```

Comment:
I'm a bit confused why 'signedRecipeUrls' implies it contains multiple URLs. `get` only allows you to fetch 1 at a time. 

Also note that your test server returns something for `recipe-signed` but not `recipes-signed`, from a quick look, so can you clarify the desired naming here, and what exactly this is supposed to do?